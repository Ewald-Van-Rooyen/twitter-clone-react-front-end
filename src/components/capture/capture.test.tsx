import {act} from "react-dom/test-utils";

import TestRenderer from "react-test-renderer";
import ReactTestUtils from "react-dom/test-utils";

import React from "react";
import Capture from "./capture";
import * as reactRedux from "react-redux";
import ReactDOM from "react-dom";

let container: any;
const defaultProfilePic = "https://avatarfiles.alphacoders.com/139/thumb-139523.jpg";

const activeUser = {
    user: {
        activeUser: {
            "id": 3,
            "username": "@Ironman",
            "role": "user",
            "usersDetailsId": 3,
            "profilePic": "https://avatarfiles.alphacoders.com/139/thumb-139523.jpg"
        }
    }
};

describe("Capture component tests", () => {

    const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
    const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");

    beforeEach(() => {
        useSelectorMock.mockClear();
        useDispatchMock.mockClear();
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    it("should render the Capture component", () => {
        useSelectorMock.mockReturnValue(activeUser);

        const testRenderer = TestRenderer.create(
            <Capture/>
        );
        expect(testRenderer).toMatchSnapshot();
    });


    it("should submit a tweet", () => {
        useSelectorMock.mockReturnValue(activeUser);
        const newTweet = "giraffe are cool";
        act(() => {
            ReactDOM.render(<Capture/>, container);
        });

        const input = container.querySelector("textarea");

        expect(input.value).toEqual("");

        input.value = newTweet;
        ReactTestUtils.Simulate.change(input);
        expect(input.value).toEqual(newTweet);
    });
});
