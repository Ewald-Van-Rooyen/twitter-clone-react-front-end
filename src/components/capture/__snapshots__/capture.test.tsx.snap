// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capture component tests should render the Capture component 1`] = `
<div
  className="horizontal-container capture-container sticky"
>
  <div
    className="display-flex"
  >
    <img
      alt=""
      className="avatar"
      src="undefined"
    />
  </div>
  <form
    className="tweet-form"
    onSubmit={[Function]}
  >
    <div
      className="capture-input-container"
    >
      <textarea
        className="capture-input"
        onChange={[Function]}
        placeholder="What's happening?"
        value=""
      />
    </div>
    <div
      className="tweet-button-container"
    >
      <button
        className="tweet-button twitter-blue-background"
        type="submit"
      >
        <span
          className="text"
        >
          Tweet
        </span>
      </button>
    </div>
  </form>
</div>
`;
